UnitTests/Parsing/ExpressionVisitors/TreeEvaluation/EvaluatableTreeFindingExpressionVisitorTest.cs

```diff
@@ -75,6 +75,23 @@ public void SimpleExpression_IsEvaluatable ()
     }
 
     [Test]
+    public void ExpressionTypeEnum_HasNoHoles ()
+    {
+      // We made a perf-optimization in RMLNQ-107 by avoiding calling Enum.IsDefined.
+      // This test verifies our assumption that the ExpressionType enum does not have holes.
+
+      var expected = 0;
+
+      foreach (var value in Enum.GetValues(typeof(ExpressionType)))
+      {
+        if (expected++ != (int)value)
+        {
+          Assert.Fail("Unexpected hole found in ExpressionType enum!");
```

Comment:
The assertion: please separate the effect (incrementing "expected") from the condition. We prefer to keep statements limited to one effect.  Then, please change it to an Assert.That (value, Is.EqualTo(exected), "message'). It's the more common style in our tests :)